pkg/component/component.go

```diff
@@ -470,6 +476,28 @@ func toIntermediate(policy map[string]interface{}, aliasMapping map[string]strin
 		if err != nil {
 			return nil, fmt.Errorf("invalid 'outputs.%s.log_level', %w", name, err)
 		}
+
+		// inject headers configured during enroll
+		if t == elasticsearchType && headers != nil {
```

Comment:
Why is this elasticsearch specific?

Overall I am not super happy with injecting random headers, how do we know these headers apply only to an output? Do they apply to only one? What if there are 2 elasticsearch outputs, should they both get them or only one?

Reply:
This is to support adding custom headers at enrolment time, a feature that was done with a config transformation rule prior to 8.6: https://github.com/elastic/elastic-agent/commit/69d8ae7bb3f7ec34a6f6d8b7a82f0d9bc1b86bd8

We removed this rule in 8.6 without replacing the underlying functionality and apparently broke some cloud deployments that rely on these enrolment time headers for traffic filtering.

I think the intent is that it should apply to all ES outputs.